python_modules/dagster/dagster/_core/definitions/unresolved_asset_sensor_definition.py

```diff
@@ -0,0 +1,279 @@
+        don't want the asset to materialize. So we set parent_asset_event_records to False for all
+        parents (so that if the sensor is set to materialize if any of the parents are updated, the sensor will
+        still choose to not materialize the asset) and immediately return.
+        """
+        from dagster._core.events import DagsterEventType
+        from dagster._core.storage.event_log.base import EventRecordsFilter
+
+        parent_asset_event_records = {}
+
+        for p in parent_assets:
+            print(f"Materializations for parent {p}")
+            if p in will_materialize_list:
+                # if the parent will be materialized by this sensor, then we can also
+                # materialize the child
+                parent_asset_event_records[p] = (True, None)
+            elif p in self._source_asset_keys:
+                # if the parent is a source asset, we assume it has always been updated
+                # this will be replaced if we introduce a versioning schema for source assets
+                print("parent is a source asset")
+                parent_asset_event_records[p] = (True, None)
+            else:
+                # if the parent has been materialized and is not currently being materialized, then
+                # we want to materialize the child
+
+                # get the most recent completed materialization
+                event_records = context.instance.get_event_records(
+                    EventRecordsFilter(
+                        event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                        asset_key=p,
+                        after_cursor=cursor.get(str(p)),
+                    ),
+                    ascending=False,
+                    limit=1,
+                )
+                print(f"materialization record {event_records}")
+
+                # get the most recent planned materialization
+                materialization_planned_event_records = context.instance.get_event_records(
+                    EventRecordsFilter(
+                        event_type=DagsterEventType.ASSET_MATERIALIZATION_PLANNED,
+                        asset_key=p,
+                    ),
+                    ascending=False,
+                    limit=1,
+                )
+
+                print(f"will materialize record {materialization_planned_event_records}")
+
+                if event_records and materialization_planned_event_records:
+                    # planned materialization is from a different run than the most recent
+                    # completed materialization
+                    if (
+                        event_records[0].event_log_entry.run_id
+                        != materialization_planned_event_records[0].event_log_entry.run_id
+                    ):
+                        # we don't want to materialize the child asset because one of its parents is
```

Comment:
I think it would be helpful to signify to the user somehow when an asset materialization is in progress, which is why a certain asset was skipped in execution.

Throwing out some ideas:
- Log some sort of message
- Maybe it's possible yield a skip reason when certain assets in the sensor selection are skipped?

Reply:
yeah logging would be good. i don't think we can yield a SkipReason because there is a built in check in `SensorDefinition` that only allows one SkipReason to be returned (and doesn't allow returning both a run request and a skip reason).